WooCommerce/Classes/ViewRelated/Products/Cells/ProductsTabProductTableViewCell.swift

```diff
@@ -41,6 +42,12 @@ final class ProductsTabProductTableViewCell: UITableViewCell {
         // Border color is not automatically updated on trait collection changes and thus manually updated here.
         productImageView.layer.borderColor = Colors.imageBorderColor.cgColor
     }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+
+        selectedProductImageOverlayView?.removeFromSuperview()
```

Comment:
What do you think about moving this line of code under the `else` option of line 87?
Since the `prepareForReuse()` is always called, this will remove it from the superview, also if it's not necessary.

Reply:
Good suggestion! it is easier to read when all the overlay view logic is in the same place. I updated how the overlay view is removed/configured in: https://github.com/woocommerce/woocommerce-ios/pull/2483/commits/a3420a3f52a827a85a4d0bb360e8f9c918b4bae5